src/js/ProviderManager.js

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+/*globals async, DirectoryLinksProvider, Links, PlacesProvider, gUserDatabase, gPinnedLinks, gBlockedLinks*/
+
+/**
+ * Singleton that handles link providers.
+ */
+
+ "use strict";
+
+const ProviderManager = {
+  init: async(function*(gMockFetch) {
```

Comment:
We probably should avoid mixing stuff required for testing and production code. I recently added some additional infrastructure to Karma to allows us to test erroneous responses (including cross-origin responses). It's totally fine if you want to have this stuff here during development tho - but let's work towards removing it. Happy to help with that if you let me know what you need (I'll check the tests too and see if I can get Karma to do what we need for this).